eZ/Bundle/EzPublishCoreBundle/EventListener/ContentDownloadRouteReferenceListener.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\EventListener;
+
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\Field;
+use eZ\Publish\Core\MVC\Symfony\Event\RouteReferenceGenerationEvent;
+use eZ\Publish\Core\MVC\Symfony\MVCEvents;
+use InvalidArgumentException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+class ContentDownloadRouteReferenceListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return array(
+            MVCEvents::ROUTE_REFERENCE_GENERATION => 'onRouteReferenceGeneration'
+        );
+    }
+
+    /**
+     * @throws \InvalidArgumentException If the required arguments are not correct
+     */
+    public function onRouteReferenceGeneration( RouteReferenceGenerationEvent $event )
+    {
+        $routeReference = $event->getRouteReference();
+
+        if ( $routeReference->getRoute() != 'content_download' )
+        {
+            return;
+        }
+
+        if ( !$routeReference->has( 'content' ) || !$routeReference->has( 'field_identifier' ) )
+        {
+            throw new InvalidArgumentException(
+                "A content_download route reference requires the 'content' and 'field_identifier' parameters"
+            );
+        }
+
+        $content = $routeReference->get( 'content' );
+        if ( !$content instanceof Content )
+        {
+            throw new InvalidArgumentException( "The 'content' parameter must be a Content" );
+        }
+        $routeReference->remove( 'content' );
+
+        $fieldIdentifier = $routeReference->get( 'field_identifier' );
+        $field = $content->getField( $fieldIdentifier );
```

Comment:
This will return the field in main language. You should use the TranslationHelper like in [Twig FieldRenderingExtension](https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/MVC/Symfony/Templating/Twig/Extension/FieldRenderingExtension.php#L120).